src/jsonata.js

```diff
@@ -955,9 +950,9 @@ var jsonata = (function() {
             }
         }
 
-        // iterate over the groups to evaluate the 'value' expression
-        for (key in groups) {
-            entry = groups[key];
+         // iterate over the groups to evaluate the 'value' expression
+         let generators = await Promise.all(Object.keys(groups).map(async (key, idx) => {
```

Comment:
sorry, could we push the indent back a bit to match the original?

Reply:
On it